multiprocessing_parse.py

```diff
@@ -21,42 +21,59 @@ def compose_trigger_list(files, plot_data):
         """
         """
         # TODO: FIND A DRIVER NAMES AND IMPLEMENT THEM IF THERE ARE OTHERS
-        match_string = '(^.*)(Oxs_MinDriver-Magnetization-)([0-9]{2})(-)(.*)(.omf)'
+        driver_class = 'MinDriver'
+        match_string = '(^.*)(Oxs_' + driver_class + \
+                            '-Magnetization-)([0-9]{2})(-)(.*)(.omf)'
         regex = re.compile(match_string)
         st = []
         # probe file
         filename = files[0]
         column_name = None
         try:
             m = regex.search(os.path.basename(filename))
-            column_name = 'MinDriver::Iteration'
+            if m is None:
+                raise AttributeError
+            column_name = driver_class +'::Iteration'
         except AttributeError:
-            match_string = '(^.*)(Oxs_TimeDriver-Magnetization-)([0-9]{2})(-)(.*)(.omf)'
-            column_name = 'TimeDriver::Iteration'
+            driver_class = 'TimeDriver'
+            match_string = '(^.*)(Oxs_' + driver_class + \
+                                '-Magnetization-)([0-9]{2})(-)(.*)(.omf)'
+            column_name = driver_class +'::Iteration'
             regex = re.compile(match_string)
         for filename in files:
             m = regex.search(os.path.basename(filename))
-            st.append(int(m.groups()[4]))
-        return plot_data.index[plot_data[column_name].isin(st)]
+            if m is not None:
+                st.append(int(m.groups()[4]))
+            else:
+                print(filename)
+        trigger_list = plot_data.index[plot_data[column_name].isin(st)]
+        try:
+            assert len(files) == len(trigger_list)
+        except AssertionError:
+            # duplicates appeared, take first and drop rest
+            unique_stages = plot_data[column_name][~plot_data[column_name].duplicated(keep='first')]
+            trigger_list = unique_stages.index[unique_stages.isin(st)]
+        return trigger_list
 
     @staticmethod
     def guess_file_type(directory):
-        supported_extensions = ['.omf', '.ovf']
+        supported_extensions = [('.omf', '*.odt'), ('.ovf', '.notsuppyet')]
```

Comment:
.notsuppyet? 🥇

Reply:
It is supported in latest pull. I did not mix issues